### PR TITLE
Add run descheduler as deployment

### PR DESCRIPTION
**Pull request by audip**
Changes:
- Add kubernetes YAML files to run descheduler as a long running `Deployment` resource
- Update README to include instructions to deploy descheduler using `kubectl apply` and `kustomize build`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +57,7 @@ Table of Contents
 
 ## Quick Start
 
-The descheduler can be run as a Job or CronJob inside of a k8s cluster. It has the
+The descheduler can be run as a Job or CronJob or Deployment inside of a k8s cluster. It has the
```
Comment (lixiang233) [inline README.md:60]: The descheduler can be run as a Job, CronJob, or Deployment inside of a k8s cluster.

**Review by lixiang233 — COMMENTED** [review]
/ok-to-test

File: kubernetes/deployment/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: descheduler
+  namespace: kube-system
+  labels:
+    app: descheduler
+spec:
+    replicas: 1
```
Comment (damemi) [inline kubernetes/deployment/deployment.yaml:9]: the indentation seems off for everything under `spec`, could you double check that? Maybe it's just how github is showing it

**Review by damemi — COMMENTED** [review]
looks pretty good to me, just the nits
I wonder if as a future todo we could improve the helm chart to give the option of running as a Job or Deployment?

File: kubernetes/deployment/deployment.yaml (hunk 2 above)
Comment (audip) [reply kubernetes/deployment/deployment.yaml:9]: This is a good catch, didn't realize the indentation was off

File: README.md (hunk 1 above)
Comment (audip) [reply README.md:60]: Fixing this!

**Comment by audip** [comment]
@damemi I like that suggestion of updating the helm chart to allow deploying descheduler as Cronjob or Deployment and I'll try to submit that as a separate PR.

**Comment by praxist** [comment]
hey @audip, not sure if you started on the chart yet but my org had a requirement for running this in helm as a deployment so I went ahead and made the PR for that.

**Comment by audip** [comment]
@praxist thank you for contributing that, I had not gotten around to working on it. :)

**Pull request merged**